frontend/src/metabase/query_builder/components/dataref/DataReference.tsx

```diff
@@ -1,14 +1,46 @@
-/* eslint "react/prop-types": "warn" */
-import PropTypes from "prop-types";
 import { useCallback } from "react";
 
+import type Field from "metabase-lib/v1/metadata/Field";
+import type Schema from "metabase-lib/v1/metadata/Schema";
+import type { CardId, ConcreteTableId, DatabaseId } from "metabase-types/api";
+
 import DatabasePane from "./DatabasePane";
 import FieldPane from "./FieldPane";
 import MainPane from "./MainPane";
 import { QuestionPane } from "./QuestionPane";
 import SchemaPane from "./SchemaPane";
 import { TablePane } from "./TablePane";
 
+type DatabaseItem = {
+  id: DatabaseId;
+};
+
+type SchemaItem = Schema;
+
+type TableItem = {
+  id: ConcreteTableId;
+};
+
+type QuestionItem = {
+  id: CardId;
+};
+
+type FieldItem = Field;
+
+type PageType = string;
```

Comment:
This should help avoid the cast in line 78

```suggestion
type PageType = keyof typeof PANES;
```